src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -209,6 +209,22 @@ public void FormulaCantRedefine()
                engine.SetFormula("A", "3", OnUpdate));
         }
 
+        [Fact]
+        public void VariableVersusFormula()
+        {
+            var engine = new RecalcEngine();
+            engine.SetFormula("A", "4", OnUpdate);
+            AssertUpdate("A-->4;");
+            engine.SetFormula("B", "A*10", OnUpdate);
+            AssertUpdate("B-->40;");
+
+            engine.UpdateVariable("A", FormulaValue.New(40));
+            AssertUpdate("A-->40;");
+            AssertUpdate("B-->400;");

```

Comment:
This looks like it will fail given current implementation of AssertUpdate/OnUpdate.  AssertUpdate clears the buffer, so it can only be called once. You may need to combine these into 1 line, `AssertUpdate("A-->40;B-->400;");`, although that may beg sorting issues.

Reply:
Yep it does fail, but with an actual update of (A-->4). The test fails even when I change the line(222) to AssertUpdate("A-->40;B-->400;");. A's value does not seem to change from its initial formula value.